Readme update (#1207)
deephaven/deephaven-core: README.md

| Language      | Server Application | Client Application (OpenAPI) |
| ------------- | ------------------ | ---------------------------- |
| Python        | Yes                | Yes                           |
| Java / Groovy | Yes                | Yes                           |
| JavaScript    | No                 | Yes                          |
| gRPC          | -                  | Yes                          |

## Run Deephaven

This section is a quick start guide for running Deephaven from pre-built images.  Almost all users will want to run Deephaven using pre-built images.  It is the easiest way to deploy.  For detailed instructions, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built).

Developers interested in tinkering with and modifying source code should build from the source code.  For detailed instructions on how to do this, see [Build and launch Deephaven](https://deephaven.io/core/docs/tutorials/launch-build).

> If you are not sure which of the two is right for you, you want to use the pre-built images.

Reviewer: ```suggestion
If you are not sure which of the two is right for you, use the pre-built images.
```

### Required Dependencies

Running Deephaven requires a few software packages.

> | Package | Version | OS  |
> | ------- | ------- | --- |
> | docker  | ≥19.3 | All |
> | docker-compose | ≥1.29 | All |
> | WSL | 2 | Only Windows |

Author: ```suggestion
| Package        | Version  | OS           |
| -------------- | -------- | ------------ |
| docker         | ^20.10.8 | All          |
| docker-compose | ^1.29.0  | All          |
| WSL            | 2        | Only Windows |
```

You can check if these packages are installed and functioning by running:
```
docker version
docker-compose version
docker run hello-world
```

> On Windows, these commands must be run using WSL 2.

Author: ```suggestion
On Windows, these commands must be run inside a WSL 2 terminal.
```

If you are not sure which of the two is right for you, you want to use the pre-built images.

### Required Dependencies

Running Deephaven requires a few software packages.

| Package | Version | OS  |
| ------- | ------- | --- |
| docker  | ≥19.3 | All |
| docker-compose | ≥1.29 | All |
| WSL | 2 | Only Windows |

You can check if these packages are installed and functioning by running:
```
docker version
docker-compose version
docker run hello-world
```

On Windows, these commands must be run using WSL 2.

If any dependencies are missing or unsupported versions are installed, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built/#prerequisites) for installation instructions.

### Create deployment

A directory must be created to store files and mount points for your deployment.  Here, we are using the `deephaven-deployment` directory.  

You will need to `cd` into the deployment directory to launch or interact with the deployment.

```bash
mkdir deephaven-deployment
cd deephaven-deployment
```

> :warning: **Commands in the following sections for interacting with a deployment must be run from the deployment directory.**

Author: ```suggestion
> :warning: **Commands in the following sections for interacting with a deployment must be run from the deployment directory.**
```

### Launch: Python

Run the following commands to launch Deephaven for Python server applications. 

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/python/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

### Launch: Python with example data

Run the following commands to launch Deephaven for Python server applications, with example data.

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/python-examples/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

### Launch: Groovy / Java

Run the following commands to launch Deephaven for Groovy / Java server applications. 

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/groovy/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

> These commands must be run from the deployment directory.

Author: ```suggestion
```

| Package | Version | OS  |
| ------- | ------- | --- |
| docker  | ≥19.3 | All |
| docker-compose | ≥1.29 | All |
| WSL | 2 | Only Windows |

You can check if these packages are installed and functioning by running:
```
docker version
docker-compose version
docker run hello-world
```

On Windows, these commands must be run using WSL 2.

If any dependencies are missing or unsupported versions are installed, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built/#prerequisites) for installation instructions.

### Create deployment

A directory must be created to store files and mount points for your deployment.  Here, we are using the `deephaven-deployment` directory.  

You will need to `cd` into the deployment directory to launch or interact with the deployment.

```bash
mkdir deephaven-deployment
cd deephaven-deployment
```

### Launch: Python

Run the following commands to launch Deephaven for Python server applications. 

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/python/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

> These commands must be run from the deployment directory.

Author: ```suggestion
```

### Launch: Python with example data

Run the following commands to launch Deephaven for Python server applications, with example data.

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/python-examples/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

> These commands must be run from the deployment directory.
> 

Author: ```suggestion
```

### Launch: Groovy / Java

Run the following commands to launch Deephaven for Groovy / Java server applications. 

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/groovy/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

These commands must be run from the deployment directory.

### Launch: Groovy / Java with example data

Run the following commands to launch Deephaven for Groovy / Java server applications, with example data.

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/groovy-examples/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

> These commands must be run from the deployment directory.
> 

Author: ```suggestion
```

### Monitor logs

The `-d` option to `docker-compose` causes the containers to run in the background, in detached mode.  This option allows you to use your shell after Docker launches the containers.

Since the container is running detached, you will not see any logs. However, you can follow the logs by running:

```bash
docker-compose logs -f
```

> This command must be run from the deployment directory.
> 

Author: ```suggestion
```

Use CTRL+C to stop monitoring the logs and return to a prompt.

### Shutdown

The deployment can be brought down by running:

```bash
docker-compose down
```

This command must be run from the deployment directory.

### Manage example data

[Deephaven's examples repository](https://github.com/deephaven/examples) contains data sets that are useful when learning 
to use Deephaven. These data sets are used extensively in Deephaven's documentation and are needed to run some examples. [Deephaven's examples repository](https://github.com/deephaven/examples) contains documentation on the available data sets and how to manage them. 

If you have chosen a deployment with example data, the example data sets will be downloaded. Production deployments containing your own data will not need the example data sets.

To upgrade a deployment to the latest example data, run:

```bash
docker-compose run examples download
```

To see what other example data management commands are available, run:

```bash
docker-compose run examples
```

> These commands must be run from the deployment directory.  If your deployment does not have example data, these commands will fail with `ERROR: No such service`.

Author: ```suggestion
If your deployment does not have example data, these commands will fail with `ERROR: No such service`.
```

If you are not sure which of the two is right for you, you want to use the pre-built images.

### Required Dependencies

Running Deephaven requires a few software packages.

| Package | Version | OS  |
| ------- | ------- | --- |
| docker  | ≥19.3 | All |
| docker-compose | ≥1.29 | All |
| WSL | 2 | Only Windows |

You can check if these packages are installed and functioning by running:
```
docker version
docker-compose version
docker run hello-world
```

On Windows, these commands must be run using WSL 2.

If any dependencies are missing or unsupported versions are installed, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built/#prerequisites) for installation instructions.

### Create deployment

A directory must be created to store files and mount points for your deployment.  Here, we are using the `deephaven-deployment` directory.  

You will need to `cd` into the deployment directory to launch or interact with the deployment.

```bash
mkdir deephaven-deployment
cd deephaven-deployment
```

> :warning: **Commands in the following sections must be run from the deployment directory.**

Author: ```suggestion
:warning: **Commands in the following sections for interacting with a deployment must be run from the deployment directory.**
```

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/groovy/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

These commands must be run from the deployment directory.

### Launch: Groovy / Java with example data

Run the following commands to launch Deephaven for Groovy / Java server applications, with example data.

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/groovy-examples/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

These commands must be run from the deployment directory.

### Monitor logs

The `-d` option to `docker-compose` causes the containers to run in the background, in detached mode.  This option allows you to use your shell after Docker launches the containers.

Since the container is running detached, you will not see any logs. However, you can follow the logs by running:

```bash
docker-compose logs -f
```

This command must be run from the deployment directory.

Use CTRL+C to stop monitoring the logs and return to a prompt.

### Shutdown

The deployment can be brought down by running:

```bash
docker-compose down
```

> This command must be run from the deployment directory.
> 

Author: ```suggestion
```

Developers interested in tinkering with and modifying source code should build from the source code.  For detailed instructions on how to do this, see [Build and launch Deephaven](https://deephaven.io/core/docs/tutorials/launch-build).

If you are not sure which of the two is right for you, you want to use the pre-built images.

### Required Dependencies

Running Deephaven requires a few software packages.

| Package | Version | OS  |
| ------- | ------- | --- |
| docker  | ≥19.3 | All |
| docker-compose | ≥1.29 | All |
| WSL | 2 | Only Windows |

You can check if these packages are installed and functioning by running:
```
docker version
docker-compose version
docker run hello-world
```

On Windows, these commands must be run using WSL 2.

If any dependencies are missing or unsupported versions are installed, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built/#prerequisites) for installation instructions.

### Create deployment

> A directory must be created to store files and mount points for your deployment.
> Here, we are using the `deephaven-deployment` directory.  

Author: ```suggestion
A directory must be created to store files and mount points for your deployment.  Here, we are using the `deephaven-deployment` directory.  
```

You will need to `cd` into the deployment directory to launch or interact with the deployment.

```bash
mkdir deephaven-deployment
cd deephaven-deployment
```

> 

Author: ```suggestion

:warning: **Commands in the following sections must be run from the deployment directory.**
```

docker-compose up -d
```

These commands must be run from the deployment directory.

### Launch: Python with example data

Run the following commands to launch Deephaven for Python server applications, with example data.

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/python-examples/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

These commands must be run from the deployment directory.

### Launch: Groovy / Java

Run the following commands to launch Deephaven for Groovy / Java server applications. 

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/groovy/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

These commands must be run from the deployment directory.

### Launch: Groovy / Java with example data

Run the following commands to launch Deephaven for Groovy / Java server applications, with example data.

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/groovy-examples/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

These commands must be run from the deployment directory.

### Monitor logs

> The `-d` option to `docker-compose` causes the containers to run in the background, in detached mode. 
> This option allows you to use your shell after Docker launches the containers.

Author: ```suggestion
The `-d` option to `docker-compose` causes the containers to run in the background, in detached mode.  This option allows you to use your shell after Docker launches the containers.
```

```bash
docker-compose logs -f
```

This command must be run from the deployment directory.

Use CTRL+C to stop monitoring the logs and return to a prompt.

### Shutdown

The deployment can be brought down by running:

```bash
docker-compose down
```

This command must be run from the deployment directory.

### Manage example data

[Deephaven's examples repository](https://github.com/deephaven/examples) contains data sets that are useful when learning 
to use Deephaven. These data sets are used extensively in Deephaven's documentation and are needed to run some examples. [Deephaven's examples repository](https://github.com/deephaven/examples) contains documentation on the available data sets and how to manage them. 

If you have chosen a deployment with example data, the example data sets will be downloaded. Production deployments containing your own data will not need the example data sets.

To upgrade a deployment to the latest example data, run:

```bash
docker-compose run examples download
```

To see what other example data management commands are available, run:

```bash
docker-compose run examples
```

> These commands must be run from the deployment directory.  
> If your deployment does not have example data, these commands will fail with `ERROR: No such service`.

Author: ```suggestion
These commands must be run from the deployment directory.  If your deployment does not have example data, these commands will fail with `ERROR: No such service`.
```

| Language      | Server Application | Client Application (OpenAPI) |
| ------------- | ------------------ | ---------------------------- |
| Python        | Yes                | Yes                           |
| Java / Groovy | Yes                | Yes                           |
| JavaScript    | No                 | Yes                          |
| gRPC          | -                  | Yes                          |

## Run Deephaven

> This section is a quick start guide for running Deephaven from pre-built images.
> For detailed instructions on running Deephaven, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built).

Author: ```suggestion
This section is a quick start guide for running Deephaven from pre-built images.  Almost all users will want to run Deephaven using pre-built images.  For detailed instructions, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built).
```

> For detailed instructions on building and running Deephaven from source code, 
> Developers interested in tinkering with and modifying source code should build from the source code.  For detailed instructions on how to do this, see [Build and launch Deephaven](https://deephaven.io/core/docs/tutorials/launch-build).

Author: ```suggestion
Developers interested in tinkering with and modifying source code should build from the source code.  For detailed instructions on how to do this, see [Build and launch Deephaven](https://deephaven.io/core/docs/tutorials/launch-build).
```

docker-compose pull
docker-compose up -d
```

These commands must be run from the deployment directory.

### Monitor logs

The `-d` option to `docker-compose` causes the containers to run in the background, in detached mode. 
This option allows you to use your shell after Docker launches the containers.

Since the container is running detached, you will not see any logs. However, you can follow the logs by running:

```bash
docker-compose logs -f
```

This command must be run from the deployment directory.

Use CTRL+C to stop monitoring the logs and return to a prompt.

### Shutdown

The deployment can be brought down by running:

```bash
docker-compose down
```

This command must be run from the deployment directory.

### Manage example data

[Deephaven's examples repository](https://github.com/deephaven/examples) contains data sets that are useful when learning 
to use Deephaven. These data sets are used extensively in Deephaven's documentation and are needed to run some examples. [Deephaven's examples repository](https://github.com/deephaven/examples) contains documentation on the available data sets and how to manage them. 

> If you have chosen a deployment with example data, the example data sets will be downloaded. Production deployments containing 
> your own data will not need the example data sets.

Author: ```suggestion
If you have chosen a deployment with example data, the example data sets will be downloaded. Production deployments containing your own data will not need the example data sets.
```

| Language      | Server Application | Client Application (OpenAPI) |
| ------------- | ------------------ | ---------------------------- |
| Python        | Yes                | Yes                           |
| Java / Groovy | Yes                | Yes                           |
| JavaScript    | No                 | Yes                          |
| gRPC          | -                  | Yes                          |

## Run Deephaven

This section is a quick start guide for running Deephaven from pre-built images.

> For detailed instructions on running Deephaven, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built).

Reviewer: ```suggestion
For detailed instructions on running Deephaven, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built). The pre-built images are for normal Deephaven deployments.
```

Reviewer 2: Are we sure most users will know what "normal Deephaven deployments" are?  This feels awkward when coupled with the line below that says "if you're not sure, use this one."  Does this work better?

```suggestion
Standard users should build and run Deephaven from pre-built images.  For detailed instructions, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built).
```

Author: I think "standard users" has a similar problem.  Ugh.

For detailed instructions on building and running Deephaven from source code, 

> see [Build and launch Deephaven](https://deephaven.io/core/docs/tutorials/launch-build).

Reviewer: ```suggestion
see [Build and launch Deephaven](https://deephaven.io/core/docs/tutorials/launch-build). Building from source is for developers that are modifying the Deephaven Community Core source code.

```

Reviewer: "see" needs to join the line above - I'm unable to remove that line break in suggestion mode.

Reviewer 2: The structure would be better to have the intended audience followed by the link.

```suggestion
Developers interested in tinkering with and modifying source code should build from the source code.  For detailed instructions on how to do this, see [Build and launch Deephaven](https://deephaven.io/core/docs/tutorials/launch-build).
```

If you are not sure which of the two is right for you, you want to use the pre-built images.

> The pre-built images are for normal Deephaven deployments.

Reviewer: ```suggestion
```

Author: What is your thinking on removing this?

Author: I see it is moved below.

> Building from source is for developers that are modifying the Deephaven Community Core source code.

Reviewer: ```suggestion
```

### Required Dependencies

Running Deephaven requires a few software packages.

| Package | Version | OS  |
| ------- | ------- | --- |
| docker  | ≥19.3 | All |

> | docker-compose | ≥1.29 | All |

Reviewer 3: I wouldn't be against using semver ranges if we need to be more specific in the versions. Probably not needed at this point though unless we know we can get value from them

https://www.telerik.com/blogs/the-mystical-magical-semver-ranges-used-by-npm-bower

Author: The problem is that we don't know the ranges. Right now, we only know that if you use versions less than this things don't work.

Reviewer 3: Gotcha, works for me

| WSL | 2 | Only Windows |

You can check if these packages are installed and functioning by running:
```
docker version
docker-compose version
docker run hello-world
```

On Windows, these commands must be run using WSL 2.

If any dependencies are missing or unsupported versions are installed, 

> see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built).

Reviewer 3: ```suggestion
see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built/#prerequisites) for installation instructions.
```
Small wording change to emphasize that installation instructions are here. Also linking to the specific header may make it more clear

Reviewer: also remove the extra line break, please

> If any dependencies are missing or unsupported versions are installed, 
> see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built).

Author: ```suggestion
If any dependencies are missing or unsupported versions are installed, see [Launch Deephaven from pre-built images](https://deephaven.io/core/docs/tutorials/launch-pre-built/#prerequisites) for installation instructions.
```

> ### Create Deployment

Reviewer: ```suggestion
### Create deployment
```

A directory must be created to store files and mount points for your deployment.
Here, we are using the `deephaven-deployment` directory.  

> You will need to cd into the deployment directory to launch or interact with the deployment.

Reviewer: ```suggestion
You will need to `cd` into the deployment directory to launch or interact with the deployment.
```

```bash
mkdir deephaven-deployment
cd deephaven-deployment
```

> ### Launch: Python

Reviewer 3: ```suggestion
#### Launch: Python
```
Would it make sense to nest these launch commands underneath the `Create Deployment` header? We could also put the `These commands must be run from the deployment directory.` statement right before this header in order to not need to copy-paste that everywhere. No preference either way to me, just a thought I had

Author: I intentionally did not put them there.  Creating and launching are not the same.  Additionally, all of the following commands need to be run from the deployment director.  This includes things that are not just launch.

Reviewer 3: Ah ok, that makes sense

Author: I'm not sure what to do with the `These commands must be run from the deployment directory.` line.  I think it is reasonable to put it in the `Create deployment` section, but I'm concerned that if it is only there, we will have users try to run the commands from other places without carefully reading the whole document.  We basically clean things up at the expense of potential user problems.

Author: I think I have an acceptable solution in.

Run the following commands to launch Deephaven for Python server applications. 

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/python/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

> 

Reviewer 3: ```suggestion

These commands must be run from the deployment directory.

```

This section doesn't have this statement while the others do

> ### Launch: Python With Example Data

Reviewer: ```suggestion
### Launch: Python with example data
```

Run the following commands to launch Deephaven for Python server applications, with example data.

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/python-examples/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

These commands must be run from the deployment directory.

### Launch: Groovy / Java

Run the following commands to launch Deephaven for Groovy / Java server applications. 

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/groovy/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

These commands must be run from the deployment directory.

> ### Launch: Groovy / Java With Example Data

Reviewer: ```suggestion
### Launch: Groovy / Java with example data
```

Run the following commands to launch Deephaven for Groovy / Java server applications, with example data.

```bash
curl https://raw.githubusercontent.com/deephaven/deephaven-core/main/containers/groovy-examples/docker-compose.yml -O
docker-compose pull
docker-compose up -d
```

These commands must be run from the deployment directory.

> ### Monitor Logs

Reviewer: ```suggestion
### Monitor logs
```

The `-d` option to `docker-compose` causes the containers to run in the background, in detached mode. 
This option allows you to use your shell after Docker launches the containers.

Since the container is running detached, you will not see any logs. However, you can follow the logs by running:

```bash
docker-compose logs -f
```

This command must be run from the deployment directory.

Use CTRL+C to stop monitoring the logs and return to a prompt.

### Shutdown

The deployment can be brought down by running:

```bash
docker-compose down
```

This command must be run from the deployment directory.

### Manage example data

> The [Deephaven examples repository](https://github.com/deephaven/examples) contains data sets that are useful when learning 

Reviewer: ```suggestion
[Deephaven's examples repository](https://github.com/deephaven/examples) contains data sets that are useful when learning 
```

> to use Deephaven. These data sets are used extensively in Deephaven's documentation and are needed to run some examples.

Reviewer: ```suggestion
to use Deephaven. These data sets are used extensively in Deephaven's documentation and are needed to run some examples. [Deephaven's examples repository](https://github.com/deephaven/examples) contains documentation on the available data sets and how to manage them. 
```

Reviewer: extra line break

If you have chosen a deployment with example data, the example data sets will be downloaded. Production deployments containing 
your own data will not need the example data sets.

> [Deephaven's examples repository](https://github.com/deephaven/examples) contains documentation on the available data sets. 

Reviewer: ```suggestion
```

> Additionally, there is documentation on managing the data sets. This includes instructions on how to upgrade to the latest version.

Reviewer: ```suggestion
```

To upgrade a deployment to the latest example data, run:

```bash
docker-compose run examples download
```

To see what other example data management commands are available, run:

```bash
docker-compose run examples
```

These commands must be run from the deployment directory.  

> If you deployment does not have example data, these commands will fail with `ERROR: No such service`.

Reviewer: ```suggestion
If your deployment does not have example data, these commands will fail with `ERROR: No such service`.
```

## Run Deephaven IDE

Once Deephaven is running, you can launch a Deephaven IDE in your web browser.  Deephaven IDE allows you

> to interactively analyze data and develop new analytics.

Reviewer: extra line break

Reviewer 2: "Interactively analyze data and develop new analytics" is basically saying the same thing twice.

> Once Deephaven is running, you can launch a Deephaven IDE in your web browser.  Deephaven IDE allows you
> to interactively analyze data and develop new analytics.

Author: ```suggestion
Once Deephaven is running, you can launch a Deephaven IDE in your web browser.  Deephaven IDE allows you to interactively analyze data.
```

- If Deephaven is running locally,

> navigate to [http://localhost:10000/ide/](http://localhost:10000/ide/).

Reviewer: line break

> - If Deephaven is running locally,
> navigate to [http://localhost:10000/ide/](http://localhost:10000/ide/).

Author: ```suggestion
- If Deephaven is running locally, navigate to [http://localhost:10000/ide/](http://localhost:10000/ide/).
```

- If Deephaven is running remotely, navigate

> to `http://<hostname>:10000/ide/`, where `<hostname>` is the address of the machine Deephaven is running on.

Reviewer: line break

> - If Deephaven is running remotely, navigate
> to `http://<hostname>:10000/ide/`, where `<hostname>` is the address of the machine Deephaven is running on.

Author: ```suggestion
- If Deephaven is running remotely, navigate to `http://<hostname>:10000/ide/`, where `<hostname>` is the address of the machine Deephaven is running on.
```
